docs/development/pigs/pig-009.rst

```diff
@@ -0,0 +1,298 @@
+is described here_:
+
+
+The basic design of the class is as following:
+
+.. code::
+
+    from gammapy.utils.distributions import InverseCDFSampler
+
+    pdf = np.array([])
+    cdf_sampler = InverseCDFSampler(pdf=pdf, random_state=random_state)
+
+Where the `gammapy.utils.random.get_random_state` method is used. The class 
+implements mainly the `.sample()` method to draw samples from the given pdf:
+
+.. code::
+
+    cdf_sampler = InverseCDFSampler()
+    samples = cdf_sampler.sample(size=100)
+
+The returned object is a `np.ndarray` containing the sampled indices.
+
+In addition the `InverseCDFSampler` should have the possibility to sample only 
+along a given axis of the PDF. This features is required to handle sampling of 
+multiple PDFs at the same time to avoid a Python loop over the PDFs. This will 
+be supported by specifying the `axis` argument:
+
+.. code::
+
+    cdf_sampler = InverseCDFSampler(pdf=pdfs, axis=0)
+    samples = cdf_sampler.sample(size=1)
+
+
+The `InverseCDFSampler` could addtionally implement a few helper functions to plot
+the CDF and PDF. The `InverseCDFSampler` class will replace the current
+`GeneralRandom` and `GeneralRandomArray` classes in `gammapy.utils.distributions`. 
+
+
+
+MapEventSampler
+---------------
+
+To support sampling from the predicted number of counts maps for source as well 
+as background maps, we propose to introduce a `MapEventSampler` class. The 
+responsibility of this class is to handle the physical coordinate transformations, 
+time dependence of the model, output data structures and sampling of the total 
+number of observed events.
+Starting from a predicted number of counts map (e.g. computed by the `MapEvaluator`):
+
+.. code::
+
+    src = SkyModel()
+    evaluator = MapEvaluator(src, exposure, psf=None, edisp=None)
+    npred_map = evaluator.compute_npred()
+
+The `MapEventSampler` takes this map and a lightcurve model as an input and 
+allows to sample from it with the `.sample()` method:
+
+.. code::
+
+    lightcurve = LightCurveTableModel() # or PhaseCurveTableModel()
+    src_sampler = MapEventSampler(npred_map, lightcurve, t_min=, t_max=)
+
+    # draw total number of events
+    n_events = src_sampler.npred_total()
+
+    # draw events as an astropy table
+    events_src = map_sampler.sample()
+
+If no lightcurve is provided a constant rate can be assumed e.g. for 
+background models:
+
+.. code::
+
+    bkg_map = BackgroundModel().map
+    bkg_sampler = MapEventSampler(bkg_map, t_min=, t_max=)
+    events_bkg = bkg_sample.sample()
+
+
+The `MapEventSampler` returns an `astropy.table.Table` object, with the following
+columns: `TIME`, `RA_TRUE`, `DEC_TRUE`, `ENERGY_TRUE`. Later additional info such
+as `DETX` and `DETY` could be added.
+
+
+IRFEventDistributor
+-------------------
+
+We also propose a class to handle correctly the spatially varying IRFs: the
+`IRFEventDistibutor` class. Based on the `PSFMap` and `EDispMap`
```

Comment:
Suggestion to use "Sampler" also for IRFs, instead of a new term "Distributor". I understand it's just a scattering offset wrt. the true parameter that was sampled before, but still, it's a kind of sampling and I would prefer to have all classes involved have the same name to make them easier to recognise and find.